core/modules/layout/includes/layout.class.inc

```diff
@@ -788,6 +788,11 @@ class Layout {
    * @return LayoutContext[]
    */
   function getContexts($include_types = LayoutContext::USAGE_TYPE_ALL) {
+    $contexts_cache = &backdrop_static(__FUNCTION__, array());
+    if (!empty($contexts_cache[$this->name])) {
```

Comment:
I'm no expert on backdrop_static, but wouldn't work the default value NULL und isset() instead of empty() better?

Can the array rightly be empty at some point? Not sure about that, either. :wink:

Reply:
`$this->contexts` could be not set or it could be array at that point, but in either of those two cases, we would want to proceed to build the context array, so I think `empty()` is better here.

Reply:
Question: since this method is called many times, should we consider using the "advanced backdrop_static() pattern", rather than calling backdrop_static() on every call?

Reply:
> Can the array rightly be empty at some point? Not sure about that, either

Came to answer @argiepiano's question and noticed I didnt quite pay attention to this point you made. You're quite right, there isn't any chance that `$contexts_cache[$this->name]` could be NULL, since its always set at least to `array()`, so `isset()` should be fine there. Is there though a performance advantage of just `isset()` vs `empty()` though? Curious.

> Question: since this method is called many times, should we consider using the "advanced backdrop_static() pattern", rather than calling backdrop_static() on every call?

I did consider, but the docs mention if the function is called "100s of times" which this is not. I dont quite know the disadvantages of the advanced pattern, so cant be sure if there is any harm in using it here.